### PR TITLE
Add multiple analog sensors interfaces and related devices 

### PR DESCRIPTION
**Pull request by traversaro**
In particular, add `multipleanalogsensorsremapper`, `multipleanalogsensorsclient` and `multipleanalogsensorsserver` devices.

See #1526 for the related discussion.
The discussion on #1526 is still ongoing and tests are missing, but I am opening the PR so that @drdanz can already check CMake code and naming patterns.

### CODE REVIEW

**Comment by traversaro** [comment]
cc @DanielePucci

**Comment by traversaro** [comment]
> I like you updated the fakeIMU. It'll be great if it was able to provide (at least) 2 sensors for each type, so it can be used also for testing the remapping. It this is gonna be a lot of work, can be done eventually in the future.

Yep, I was thinking of getting this merged in some form to start testing on the robot, and then having a more complete testing an documentation.

File: src/devices/multipleanalogsensorsclient/MultipleAnalogSensorsClient.cpp [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+    {
+        yError("MultipleAnalogSensorsClient: missing name parameter, exiting.");
+    }
+
+    if (!config.check("local"))
+    {
+        yError("MultipleAnalogSensorsClient: missing local parameter, exiting.");
+    }
+
+    std::string remote = config.find("remote").asString().c_str();
```
[the remark this answers is not in the document]
Comment (traversaro) [reply src/devices/multipleanalogsensorsclient/MultipleAnalogSensorsClient.cpp]: Typo, thanks!

**Comment by traversaro** [comment]
I think the interfaces are now ready to be merged. A lot of polishing is still necessary, but merging at least in devel will help testing them on the robot for identifying any pain point before the next YARP release.

**Comment by traversaro** [comment]
> camel case: sometimes name are all lower case, sometime camel case, sometime other combination. Can you make the names uniform? It would be good to have at least the device name in the .ini to be camel case

I used the following convention: 
* Library names : camel case, but with the first letter lower case
* Device names: all lower case 
* Class names: camel case

I think I should be inline with the yarp convention, let me know in which case I am violating some convention. 

> I like you updated the fakeIMU. It'll be great if it was able to provide (at least) 2 sensors for each type, so it can be used also for testing the remapping. It this is gonna be a lot of work, can be done eventually in the future.

This is an improvement that I planned for a later PR (this is already quite huge!).

**Comment by barbalberto** [comment]
Ok for class names and library names.
By looking at `yarpdev list` output, all possible combinations are used for device names, there is no well-established convention. Anyway most of them are camel case with first lower letter, like:

Device "fakeAnalogSensor"
Device "fakeDepthCamera"
Device "fakeIMU"
Device "imuBosch_BNO055"
Device "embObjAnalogSensor"
Device "embObjInertials"
Device "embObjMais"
Device "embObjMotionControl"
Device "embObjMultiEnc"
Device "embObjSkin"
Device "parametricCalibrator"
Device "skinWrapper"
Device "usbCamera"
Device "faceDisplayClient"

I think in this way it easier to read/write and less error-prone, both for config files and from terminal.
However, i'm fine with the PR

**Comment by traversaro** [comment]
> Anyway most of them are camel case with first lower letter, like:

You are right, I was confused by the one used for the controlboard-related devices and the cartesian control-related devices with which I am more familiar. I would propose to settle on a convention, but I am afraid to detonate a [bike shedding](https://en.wikipedia.org/wiki/Law_of_triviality) war, so unless the mantainer @drdanz decides for a convention I will continue to use the names I was using.

**Comment by traversaro** [comment]
Test passing, one review of approval, merging.

**Pull request merged**